arangod/Cluster/ActionBase.cpp

```diff
@@ -0,0 +1,227 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2016 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Kaveh Vahedipour
+/// @author Matthew Von-Maszewski
+////////////////////////////////////////////////////////////////////////////////
+
+#include "ActionBase.h"
+#include "MaintenanceFeature.h"
+
+using namespace arangodb;
+using namespace arangodb::maintenance;
+
+const char ActionBase::KEY[]="key";
+const char ActionBase::FIELDS[]="fields";
+const char ActionBase::TYPE[]="type";
+const char ActionBase::INDEXES[]="indexes";
+const char ActionBase::INDEX[]="index";
+const char ActionBase::SHARDS[]="shards";
+const char ActionBase::DATABASE[]="database";
+const char ActionBase::COLLECTION[]="collection";
+const char ActionBase::EDGE[]="edge";
+const char ActionBase::NAME[]="name";
+const char ActionBase::ID[]="id";
+const char ActionBase::LEADER[]="theLeader";
+const char ActionBase::LOCAL_LEADER[]="localLeader";
+const char ActionBase::GLOB_UID[]="globallyUniqueId";
+const char ActionBase::OBJECT_ID[]="objectId";
```

Comment:
Why are these constants here and again in ActionDescription.h?

Reply:
Will fix

Reply:
We agreed to ignore this cleanup.

Reply:
Have put in a FIXMEMAINTENANCE for post-merge cleanup